runelite-client/src/main/java/net/runelite/client/ui/overlay/components/ImageComponent.java

```diff
@@ -52,16 +57,24 @@ public Dimension render(Graphics2D graphics)
 			return null;
 		}
 
-		graphics.drawImage(image, preferredLocation.x, preferredLocation.y, null);
+		BufferedImage toRender = scaledImage != null ? scaledImage : image;
+		graphics.drawImage(toRender, preferredLocation.x, preferredLocation.y, null);
 		final Dimension dimension = new Dimension(image.getWidth(), image.getHeight());
 		bounds.setLocation(preferredLocation);
 		bounds.setSize(dimension);
 		return dimension;
 	}
 
 	@Override
-	public void setPreferredSize(Dimension dimension)
+	public void setPreferredSize(Dimension preferredSize)
 	{
-		// Just use image dimensions for now
+		if (preferredSize != null && image != null)
```

Comment:
You could probably additionally check the image size != the preferred size. I also think that we never have a preferred size of null?

Reply:
> You could probably additionally check the image size != the preferred size

Good call on this one. I'll add it in.

> I also think that we never have a preferred size of null?

Just an extra check in case it's used in a plugin or something.